textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/FuzzyTokenPredicate.java

```diff
@@ -40,33 +43,62 @@ public FuzzyTokenPredicate(String query, List<Attribute> attributeList, Analyzer
             this.tokens = Utils.tokenizeQuery(analyzer, query);
             this.computeThreshold();
             this.attributeList = attributeList;
-            this.fields = this.extractSearchFields();
+            this.extractSearchFields();
             this.luceneQuery = this.createLuceneQueryObject();
         } catch (Exception e) {
             e.printStackTrace();
             throw new DataFlowException(e.getMessage(), e);
         }
     }
-    
-    private String[] extractSearchFields() {
-    	return null;
+    public boolean getisSpanInformationAdded(){
+    	return this.isSpanInformationAdded;
+    }
+    public List<Attribute> getAttributeList() {
+    	return this.attributeList;
+    }
+    public int getThreshold() {
+    	return this.threshold;
+    }
+    public ArrayList<String> getQueryTokens(){
+    	return this.tokens;
+    }
+    private void extractSearchFields() {
+    	this.fields = new String[this.attributeList.size()];
+    	int i = 0;
+    	for( Attribute a : this.attributeList) {
+    		this.fields[i] = a.getFieldName();
+    		i++;
+    	}
 	}
 
     /*
      * The input threshold given by the end-user (thresholdRatio data member) is a ratio
      * but boolean search query requires integer as a threshold.
      */
 	public void computeThreshold() {
-    	return;
+		this.threshold = (int) (this.thresholdRatio * this.tokens.size());
+    	if(this.threshold == 0){
```

Comment:
This use case doesn't make sense.  We can either set it 1, or throw an exception.  No matter, it will be good to say in the comments of this class to set the right expectation.